### ISSUE TITLE
Custom metric sends extra screen view

### ISSUE THREAD

**Issue by LegendaryNIK** [danwilson/google-analytics-plugin#424]
Hello
I've decided to track some custom metrics in my ionic application (likes, comments, etc)

So, for example: when i track likes via trackMetric(), the 'like' count grows as I need, but it also counts as screen view in my report. And after all it shows 2 screen views and 1 like, but it should be one screen view and 1 like.
Is there any workaround for this? Or maybe I'm doing something wrong?

I track screen views this way:
`constructor(private ga:GoogleAnalytics) {this.ga.trackView(success.title,'',true).then((e) =>console.log(e))} `

And like tracking is done by function:
 `addLike(){
    this.ga.trackMetric('1', 1)
      .then((e) => {console.log(e)})
      .catch((e) => {console.log(e)});
  }`
`<button ion-button (click)="addLike()">Give a like!</button>`

PS Tried that on android build

**Comment by victorsosa** [comment]
`this.ga.trackView(success.title,'',true) ` will create a new session every time the screen is show.

`this.ga.trackView(success.title,'',false) `  ->> try using this

**Comment by LegendaryNIK** [comment]
Didn't work for me :\
For now, solved this by adding another custom metric and triggering it on ionViewWillEnter()

**Comment by victorsosa** [comment]
Got it; you can't use the constructor for trackview in Ionic 2; for this you should use ionViewWillEnter()

**Comment by LegendaryNIK** [comment]
Replaced trackview to ionViewWillEnter().
Still every trackMetric adds extra screen view.

**Comment by spacepope** [comment]
Tracking a hit should even work in the constructor of an Ionic 2 page or component - just make sure the platform ready event has fired AND your tracker instance has started before:
```javascript
constructor(private platform: Platform, private ga: GoogleAnalytics) {
   platform.ready().then(() => {
      // do this initialization only once, f.e. in a service or app component constructor..
      return ga.startTrackerWithId("MY_ID");
   }).then(() => {
      return ga.trackMetric('1', 1);
   }).catch((err) => {
      console.error(err);
   });
}
```

Regarding the extra view hit:
As stated in the [Google Analytics docs for custom metrics](https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets) 
> In some cases, you might want to send a custom dimension or metric with all the hits on a given page (or for the lifetime of the tracker object). In this case, you can set the custom dimension or metrics using the set command

That means, normally custom metrics are metadata added to a hit like a page view, an event, etc...
As it seems in the Android implementation the trackMetric method utilizes HitBuilders.ScreenViewBuilder to build a view hit and attach the custom metric, whereas (someone correct me if i get it wrong) in the iOS and Browser implementation the metric is only set on the tracker object without actually sending a hit.

So i would propose to change the Android behaviour to the same as on iOS and Browser in a future update. From usability point of view it would be best to rename trackMetric to addCustomMetric (because thats what it actually does on iOS and Browser), but that would be a breaking change..

**Comment by victorsosa** [comment]
@spacepope thanks for the inside

**Comment by victorsosa** [comment]
fixed on v1.9.0
